src/Storage.cpp

```diff
@@ -2518,6 +2521,8 @@ void Storage::addBlock(PreProcessedBlockPtr ppb, bool saveUndo, unsigned nReserv
             dspsubsmgr->enqueueNotifications(std::move(notify->dspTxsAffected));
         if (txsubsmgr && !notify->txidsAffected.empty())
             txsubsmgr->enqueueNotifications(std::move(notify->txidsAffected));
+        if (shtxsubsmgr && !notify->shtxAffected.empty())
+            shtxsubsmgr->enqueueNotifications(std::move(notify->shtxAffected));
```

Comment:
If submitted only from `MempoolSynchTask`, it is a copy of `scriptHashesAffected`.
I do not enqueue notifications from `addBlock` method. I think it might lead to the doublication of notifications - both from mempool and upon block confirmation.